cobalt/configuration/configuration.cc

```diff
@@ -0,0 +1,42 @@
+// Copyright 2025 The Cobalt Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "cobalt/configuration/configuration.h"
+
+#include "base/memory/singleton.h"
+#include "starboard/system.h"
+
+namespace cobalt {
+namespace configuration {
+
+// static
+Configuration* Configuration::GetInstance() {
+  return base::Singleton<Configuration,
+                         base::LeakySingletonTraits<Configuration>>::get();
+}
+
+Configuration::Configuration() {
+  configuration_api_ = static_cast<const CobaltExtensionConfigurationApi*>(
+      SbSystemGetExtension(kCobaltExtensionConfigurationName));
```

Comment:
Where does access to SbSystem...() functions take place? At a guess I'd say this should happen in the browser process, and perhaps in the UI thread (if it can block, possibly in the IO thread). If so then this class' methods should be protected with `DCHECK_CURRENTLY_ON(content::BrowserThread::UI);` (or `...::IO);`.

I see this method is being called directly from a Blink Object, which lives in the main Renderer thread; it would trigger that assertion. We'll need a Mojo interface to access this guy in all likelihood.

@andrewsavage1 FYI

Reply:
Isn't it being called from h5vcc_system_impl.cc (i.e. going through mojo)?